### ISSUE TITLE
Possible issue validating JSON Schema instances against metaschemas

### ISSUE THREAD

**Issue by MikeRalphson** [jdesrosiers/json-schema#1]
Forgive me if I'm doing something wrong here.

I've got `json-schema` working fine for validating arbitrary JSON instances against a schema. In my case the instances are OAS documents and the schema is the published OAS 3.0 schema.

When I try to validate the published OAS schema as the instance, using the JSON Schema metaschema as the schema, I get the following error:-

```
TypeError: Cannot read property 'compile' of undefined
      at compileSchema (/home/mike/node/openapi-specification/node_modules/@hyperjump/json-schema-core/lib/core.js:122:55)
      at compileSchema (/home/mike/node/openapi-specification/node_modules/@hyperjump/json-schema-core/lib/core.js:110:35)
      at process._tickCallback (internal/process/next_tick.js:68:7)
      at Function.Module.runMain (internal/modules/cjs/loader.js:745:11)
      at startup (internal/bootstrap/node.js:283:19)
      at bootstrapNodeJSCore (internal/bootstrap/node.js:743:3)
```

If I change any character in the `id` value of the metaschema so it doesn't collide with the actual id of the published metaschema, the error goes away and both positive and negative validations work correctly.

**Comment by jdesrosiers** [comment]
Can you link me to the OAS schema. I can't seem to find it.

**Comment by MikeRalphson** [comment]
https://raw.githubusercontent.com/OAI/OpenAPI-Specification/master/schemas/v3.0/schema.json

**Comment by jdesrosiers** [comment]
I figured it out. It's getting confused about getting a meta schema as a schema. I didn't think to test that because it's always validating against the meta schema. Turns out the order matters. I should have it fixed soon.

**Comment by jdesrosiers** [comment]
I just pushed some fixes for this an a couple other things. Give it another try and let me know if you notice anything else that doesn't seem right.

**Comment by MikeRalphson** [comment]
Can confirm this is fixed in `0.4.2`. Many thanks!
